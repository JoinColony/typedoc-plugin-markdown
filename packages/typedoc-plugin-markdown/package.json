{
  "name": "typedoc-plugin-markdown",
  "version": "3.12.1",
  "description": "A plugin for TypeDoc that enables TypeScript API documentation to be generated in Markdown.",
  "main": "dist/index.js",
  "files": [
    "dist/"
  ],
  "scripts": {
    "lint": "eslint ./src --ext .ts",
    "prepublishOnly": "yarn run lint && yarn run build && && yarn run test",
    "build": "rm -rf dist && make-dir ./dist/resources && yarn run precompile && tsc",
    "test": "jest --colors --runInBand --updateSnapshot -t \"Comments:\"",
    "build-and-test": "yarn run build && yarn run test",
    "precompile": "handlebars ./src/resources/templates -c handlebars/runtime -f ./dist/resources/templates.js && handlebars ./src/resources/partials -c handlebars/runtime -p -f ./dist/resources/partials.js && node ./build/helpers.js",
    "demos": "yarn run build && yarn run demo:md && yarn run demo:html",
    "demo:md": "typedoc --plugin typedoc-plugin-markdown --options ../../test/typedoc-options.json --out ./demo/md",
    "demo:html": "typedoc --plugin none --options ../../test/typedoc-options.json --out ./demo/html"
  },
  "author": "Thomas Grey",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/tgreyuk/typedoc-plugin-markdown/issues"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/tgreyuk/typedoc-plugin-markdown.git",
    "directory": "packages/typedoc-plugin-markdown"
  },
  "homepage": "https://github.com/tgreyuk/typedoc-plugin-markdown/tree/master/packages/typedoc-plugin-markdown",
  "keywords": [
    "markdown",
    "typescript",
    "documentation",
    "api",
    "typedocplugin",
    "typedoc"
  ],
  "peerDependencies": {
    "typedoc": ">=0.22.0"
  },
  "dependencies": {
    "handlebars": "^4.7.7"
  }
}
