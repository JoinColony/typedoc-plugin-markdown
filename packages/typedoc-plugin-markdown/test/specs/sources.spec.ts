import * as Handlebars from 'handlebars/runtime';
import { ProjectReflection } from 'typedoc';
import { MarkdownThemeRenderContext } from 'typedoc-plugin-markdown';

const getProp = (reflection: any) => {
  const prop = reflection.findReflectionByName('prop');
  prop.sources = prop.sources.map((source) => {
    delete source.url;
    return source;
  });
  return prop;
};

describe(`Sources:`, () => {
  let bootstrap: {
    project: ProjectReflection;
    context: MarkdownThemeRenderContext;
  };
  let partial: Handlebars.TemplateDelegate;

  beforeAll(async () => {
    bootstrap = global.bootstrap('sources.ts');
    partial = global.getTemplate(bootstrap.context, 'member.sources');
  });

  test(`should display implementation of sources'`, () => {
    expect(
      global.renderTemplate(
        partial,
        getProp(bootstrap.project.findReflectionByName('SomeClass')),
      ),
    ).toMatchSnapshot();
  });

  test(`should display inherited sources'`, () => {
    expect(
      global.renderTemplate(
        partial,
        getProp(bootstrap.project.findReflectionByName('AnotherInterface')),
      ),
    ).toMatchSnapshot();
  });

  test(`should display overide sources'`, () => {
    expect(
      global.renderTemplate(
        partial,
        getProp(bootstrap.project.findReflectionByName('AnotherClass')),
      ),
    ).toMatchSnapshot();
  });
});
