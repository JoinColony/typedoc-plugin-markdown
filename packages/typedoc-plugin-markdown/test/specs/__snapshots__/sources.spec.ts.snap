// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sources: should display implementation of sources' 1`] = `
"#### Implementation of

[object Object]

#### Defined in

sources.ts:6
"
`;

exports[`Sources: should display inherited sources' 1`] = `
"#### Inherited from

[SomeClass](../classes/SomeClass.md).[prop](../classes/SomeClass.md#prop)

#### Defined in

sources.ts:2
"
`;

exports[`Sources: should display overide sources' 1`] = `
"#### Overrides

[object Object]

#### Defined in

sources.ts:10
"
`;
